apps/emqx_resource/src/emqx_resource_buffer_worker.erl

```diff
@@ -1469,18 +1469,25 @@ mark_inflight_items_as_retriable(Data, WorkerMRef) ->
 %% used to update a batch after dropping expired individual queries.
 update_inflight_item(InflightTID, Ref, NewBatch, NumExpired) ->
     _ = ets:update_element(InflightTID, Ref, {?ITEM_IDX, NewBatch}),
-    ok = dec_inflight(InflightTID, NumExpired).
+    ok = dec_inflight(InflightTID, NumExpired, update).
 
 inc_inflight(InflightTID, Count) ->
     _ = ets:update_counter(InflightTID, ?SIZE_REF, {2, Count}),
     _ = ets:update_counter(InflightTID, ?BATCH_COUNT_REF, {2, 1}),
     ok.
 
-dec_inflight(_InflightTID, 0) ->
+dec_inflight(_InflightTID, 0, update) ->
```

Comment:
nit:
feel like it's more straightforward if write it as two funcs
dec_inflight_remove
dec_inflight_update

Reply:
True, it sounds better.  Made another small fix and changed that as well.

:heavy_check_mark: